### DISCUSSION TITLE
Access Body of ClientResponse after read()

### DISCUSSION THREAD

**Discussion by sefim** [aio-libs/aiohttp#5826] [Q&A]
Hi, 
I have a common code for reading data from ClientSession.get()
the reply is a ClientResponse which I can access the json and text, however I also need to access the actual bytes, why is the _body not accessible from ClientResponse

```python
async def get(url):    
    session = ClientSession()
    async with session.get(url) as response:
        await response.read()
    return response

async def get_json():
    response = get("")
    return response.json()

async def get_text():
    response = get("")
    return response.text()

async def get_body():
    response = get("")
    return response.body # !!! not available 
```

Thanks,
Sefi

**Comment by Dreamsorcerer** [comment]
I'm not really sure what you're trying to do, but there is no `body` attribute, the body is returned from `response.read()`, I'm not sure why you are calling the method in the get method and then throwing it away... You also need to `await` on `.json()` and `.text()`...

Also, you probably shouldn't be using the response object after you've left the `with` statement. I'd recommend restructuring your code so you don't return the response object from the function.
